.github/workflows/README.md

```diff
@@ -14,6 +14,87 @@
     specific language governing permissions and limitations
     under the License.
 -->
+
+# Guidelines for Adding or Modifying Workflows
+
+On top of normal Actions workflow steps, all new CI workflows (excluding release workflows or other workflow automation) should have the following:
+
+1) A set of specific triggers
+2) An explicit checkout step
```

Comment:
Does this checkout merges the PR with the master branch? If the user code is old, the test might fail with old errors that are already fixed on master

Reply:
It does not, in that case the best path forward is for users to pull the latest contents from master.

In theory, you could do something a bit more fancy like what https://github.com/actions/checkout/issues/518 suggests, but that creates harder to follow/harder to get right branching depending on if you're doing the checkout as part of a PR or another trigger. My take is that its not worth it, especially since its easy to change down the line if this becomes a problem